paddlenlp/transformers/llama/modeling_3D_auto.py

```diff
@@ -939,15 +877,16 @@ def forward(
         if position_ids is None:
             position_ids = paddle.arange(seq_length, dtype="int64").expand((batch_size, seq_length))
             # NOTE(zhaoyingli): infer spmd does not support [seq_len] --> [batch, seq_len] in data_parallel
-            position_ids = dist.shard_tensor(position_ids, get_mesh(), [dist.Shard(0), dist.Replicate()])
+            position_ids = dist.shard_tensor(position_ids, get_mesh(), [dist.Replicate(), dist.Replicate()])
```

Comment:
why change to replicated?

Reply:
Because in static mode, infer spmd hasn't supported the case -- "[seq_len] --> [batch, seq_len]"